clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -55,12 +58,17 @@
 
   public static synchronized TachyonFileSystem get() {
     if (sTachyonFileSystem == null) {
-      sTachyonFileSystem = new TachyonFileSystem();
+      boolean enableLineage = ClientContext.getConf().getBoolean(Constants.USER_ENABLE_LINEAGE);
+      if (enableLineage) {
+        return TachyonLineageFileSystem.get();
+      } else {
+        sTachyonFileSystem = new TachyonFileSystem();
+      }
```

Comment:
I also feel it is a little bit strange to get a `TachyonLineageFileSystem` instance from a static method in `TachyonLineageFileSystem` 's base class. It might be cleaner to have a separate `FileSystemFactory`. (The downside is, though, Tachyon application code needs to call `tachyon.client.FileSystemFactory.get()` to get a FileSystem handler, may be not as clean)